sqlmesh/core/loader.py

```diff
@@ -64,6 +64,8 @@ class LoadedProject:
     excluded_requirements: t.Set[str]
     environment_statements: t.List[EnvironmentStatements]
     user_rules: RuleSet
+    model_test_metadata: t.List[ModelTestMetadata]
+    models_with_tests: t.Set[str]
```

Comment:
Why is this an attribute of the `LoadedProject`? We just need the metadata right? Then you other consumers (e.g., the context, or linter) can extract this info and use it as needed.